kosat-api/src/commonMain/kotlin/Kosat.kt

```diff
@@ -0,0 +1,156 @@
+import org.kosat.CDCL
+import org.kosat.Clause
+import org.kosat.Lit
+import org.kosat.SolveResult
+import kotlin.math.abs
+
+/**
+ * Wrapper for the KoSAT solver.
+ *
+ * All the literals and variables are in DIMACS format (positive for variables,
+ * negative for negated variables, 1-based indexing).
+ *
+ * *This class is not thread-safe.*
+ */
+class Kosat {
+    private var numberOfVariables: Int = 0
+    private var numberOfClauses: Int = 0
+
+    private var solver: CDCL = CDCL()
+
+    private var result: SolveResult? = null
+    private var model: List<Boolean>? = null
+
+    private fun invalidateResult() {
+        result = null
+        model = null
+    }
+
+    private fun checkLiterals(literals: Iterable<Int>) {
+        for (lit in literals) {
+            if (lit == 0) {
+                throw IllegalArgumentException("Clause must not contain 0")
+            } else {
+                val varIndex = abs(lit) - 1
+                if (varIndex >= numberOfVariables) {
+                    throw IllegalArgumentException(
+                        "Variable of $lit is not defined (total variables: ${numberOfVariables})"
+                    )
+                }
+            }
+        }
+    }
+
+    /**
+     * Resets the solver to its initial state, removing all variables and
+     * clauses.
+     */
+    fun reset() {
+        numberOfVariables = 0
+        numberOfClauses = 0
+        solver = CDCL()
+        invalidateResult()
+    }
+
+    /**
+     * Return the number of variables in the solver.
+     */
+    fun numberOfVariables(): Int {
+        return numberOfVariables
+    }
+
+    /**
+     * Return the number of clauses in the solver.
+     */
+    fun numberOfClauses(): Int {
+        return numberOfClauses
+    }
+
+    /**
+     * Allocate a new variable in the solver.
+     */
+    fun newVariable() {
+        invalidateResult()
+        numberOfVariables++
+        solver.newVariable()
+    }
+
+    /**
+     * Add a new clause to the solver. The literals must be in DIMACS format
+     * (positive for variables, negative for negated variables, 1-based
+     * indexing). All variables must be defined with [newVariable] before adding
+     * clauses.
+     */
+    fun newClause(clause: Iterable<Int>) {
+        checkLiterals(clause)
+        invalidateResult()
+        numberOfClauses++
+        solver.newClause(Clause.fromDimacs(clause))
+    }
+
+    /**
+     * Add a new clause to the solver. The literals must be in DIMACS format
+     * (positive for variables, negative for negated variables, 1-based
+     * indexing). All variables must be defined with [newVariable] before adding
+     * clauses.
+     */
+    fun newClause(vararg literals: Int) {
```

Comment:
What about the users that already have an IntArray of literals? Are we going to force them to use `*`-spread operator `solver.newClause(*literals)` ?